cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/MilepostController.java

```diff
@@ -308,6 +311,74 @@ public ResponseEntity<Collection<com.trihydro.cvdatacontroller.model.Milepost>>
 		return ResponseEntity.ok(data);
 	}
 
+	@RequestMapping(method = RequestMethod.POST, value="/set-milepost-cache")
+	public ResponseEntity<String> setMilepostCache(@RequestBody SetMilepostCacheRequest milepostCacheBody) {
+		
+		utility.logWithDate("Setting milepost cache for timID: " + milepostCacheBody.getTimID());
+
+		if (milepostCacheBody.getMileposts().size() == 0 || milepostCacheBody.getTimID() == null) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Invalid Request: please provide a valid milepost list and timID");
+		}
+		if (milepostCache.containsKey(milepostCacheBody.getTimID())) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Milepost cache already exists for timID: " + milepostCacheBody.getTimID());
+		}
+		milepostCache.put(milepostCacheBody.getTimID(), milepostCacheBody.getMileposts());
+		return ResponseEntity.ok("Milepost cache set successfully for timID: " + milepostCacheBody.getTimID());
+	}
+
+	@RequestMapping(method = RequestMethod.GET, value="/get-milepost-cache/{timID}")
+	public ResponseEntity<List<Milepost>> getMilepostCacheByTimID(@PathVariable String timID) {
+		List<Milepost> mileposts = new ArrayList<>();
+
+		if (milepostCache.containsKey(timID)) {
+			mileposts = milepostCache.get(timID);
+			utility.logWithDate("Found " + mileposts.size() + " mileposts in cache for timID: " + timID);
+			return ResponseEntity.ok(milepostCache.get(timID));
+		}
+
+		return ResponseEntity.ok(mileposts);
+	}
+
+	@RequestMapping(method = RequestMethod.DELETE, value="/delete-milepost-cache/{timID}")
+	public ResponseEntity<String> deleteMilepostCache(@PathVariable String timID) {
+		utility.logWithDate("Deleting milepost cache for timID: " + timID);
+
+		if (milepostCache.containsKey(timID)) {
+			milepostCache.remove(timID);
+			return ResponseEntity.ok("Milepost cache deleted successfully for timID: " + timID);
+		}
+		return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Milepost cache not found for timID: " + timID);
+	}
+
+	@RequestMapping(method = RequestMethod.GET, value="/clear-milepost-cache")
+	public ResponseEntity<String> clearMilepostCache() {
+		utility.logWithDate("Clearing milepost cache");
+		List<String> timIDs = new ArrayList<>(milepostCache.keySet());
+		List<String> activeTimIds = getActiveTimIds();
+		// remove all active TIM IDs from the list of milepost cache TIM IDs
+		timIDs.removeAll(activeTimIds);
+		for (String timID : timIDs) {
+			milepostCache.remove(timID);
+		}
+		return ResponseEntity.ok("Milepost cache cleared successfully");
+	}
+
+	private List<String> getActiveTimIds() {
+		List<String> activeTimIds = new ArrayList<>();
+		String sql = "SELECT client_id FROM active_tim WHERE marked_for_deletion = False";
```

Comment:
question: the above code references the cache via tim_id, however here we're selecting client_id?

Reply:
I've renamed these variables for clarity, as they're both referencing the clientID

Reply:
Yes, I've renamed this to `getActiveTimClientIds` for clarity